src/bosh-softlayer-cpi/softlayer/pool/softlayer_pool_creator.go

```diff
@@ -427,3 +433,39 @@ func (c *softLayerPoolCreator) oSReloadVMInPool(cid int, agentID string, stemcel
 	}
 	return vm, nil
 }
+
+func (c *softLayerPoolCreator) tagAsManage(cid int) error {
+	service, err := c.softLayerClient.GetSoftLayer_Virtual_Guest_Service()
+	if err != nil {
+		return bosherr.WrapErrorf(err, "Getting Service of SoftLayer VirtualGuest `%d`", cid)
+	}
+
+	tagReferences, err := service.GetTagReferences(cid)
+	if err != nil {
+		return bosherr.WrapErrorf(err, "Getting TagReferences on SoftLayer VirtualGuest `%d`", cid)
+	}
+
+	existsTags := make([]string, len(tagReferences), len(tagReferences)+1)
+	tag := "vps-manage"
+	found := false
+	for index, tagReference := range tagReferences {
+		if tag == tagReference.Tag.Name {
+			found = true
+			break
+		}
+		existsTags[index] = tagReference.Tag.Name
+	}
+
+	if !found {
+		success, err := service.SetTags(cid, append(existsTags, tag))
+		if !success {
+			return bosherr.WrapErrorf(err, "Setting tags on SoftLayer VirtualGuest `%d`", cid)
+		}
+
+		if err != nil {
```

Comment:
I would check for error before success since any error should be a failure. Correct?